dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,489 @@
+        The gradient table for this prediction
+    S0 : float or ndarray
+        The non diffusion-weighted signal in every voxel, or across all
+        voxels. Default: 1
+
+    Returns
+    --------
+    S : (..., N) ndarray
+        Simulated signal based on the free water DTI model
+
+    Notes
+    -----
+    The predicted signal is given by:
+    $S(\theta, b) = S_0 * [f * e^{-b ADC_{r}} + (1-f) * e^{-b ADC_{h}]$, where
+    $ ADC_{r} and ADC_{h} are the apparent diffusion coefficients of the
+    diffusion hindered and restricted compartment for a given direction
+    $\theta$, $b$ is the b value provided in the GradientTable input for that
+    direction, $f$ is the volume fraction of the restricted diffusion
+    compartment (also known as the axonal water fraction).
+    """
+
+    # Initialize pred_sig
+    pred_sig = np.zeros(f.shape + (gtab.bvals.shape[0],))
+
+    # Define dti design matrix and region to process
+    D = dti_design_matrix(gtab)
+    evals = params[..., :3]
+    mask = _positive_evals(evals[..., 0], evals[..., 1], evals[..., 2])
+
+    # Prepare parameters
+    f = params[..., 39]
+    adce = params[..., 27:33]
+    adce = params[..., 33:39]
+
+    # Process pred_sig for all data voxels
+    index = ndindex(evals.shape[:-1])
+    for v in index:
+        if mask[v]:
+            pred_sig[v] = (1 - f[v]) * np.exp(np.dot(adce[v], D.T)) + \
+                          f[v] * np.exp(np.dot(adci[v], D.T))
+
+    return pred_sig
+
+
+class KurtosisMicrostructuralModel(DiffusionKurtosisModel):
+    """ Class for the Diffusion Kurtosis Microstructural Model
+    """
+
+    def __init__(self,  gtab, fit_method="WLS", *args, **kwargs):
+        """ Initialize a KurtosisMicrostruturalModel class instance [1]_.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+
+        fit_method : str or callable
+            str can be one of the following:
+            'OLS' or 'ULLS' to fit the diffusion tensor and kurtosis tensor
+            using the ordinary linear least squares solution
+                dki.ols_fit_dki
+            'WLS' or 'UWLLS' to fit the diffusion tensor and kurtosis tensor
+            using the ordinary linear least squares solution
+                dki.wls_fit_dki
+
+            callable has to have the signature:
+                fit_method(design_matrix, data, *args, **kwargs)
+
+        args, kwargs : arguments and key-word arguments passed to the
+           fit_method. See dki.ols_fit_dki, dki.wls_fit_dki for details
+
+        Notes
+        -----
+        1) Since this model is an extension of DKI, class instance is defined
+           as subclass DiffusionKurtosisModel from dki.py
+        2) The first step of the DKI based microstructural model requires
+           diffusion tensor and kurtosis tensor fit. This fit is performed
+           using the DKI model solution specified by users by input parameter
+           fit_method.
```

Comment:
I don't think that we need these notes in the documentation. They both seem like implementation details, rather than an explanation of functionality or API.